gc/src/lib.rs

```diff
@@ -60,19 +60,43 @@ impl<T: Trace> Gc<T> {
     /// let five = Gc::new(5);
     /// ```
     pub fn new(value: T) -> Self {
+        debug_assert!(mem::align_of::<GcBox<T>>() > 1);
+
         unsafe {
             // Allocate the memory for the object
             let ptr = GcBox::new(value);
 
             // When we create a Gc<T>, all pointers which have been moved to the
             // heap no longer need to be rooted, so we unroot them.
             (**ptr).value().unroot();
-            Gc { ptr: ptr, root: Cell::new(true) }
+            let gc = Gc { ptr_root: Cell::new(NonZero::new(*ptr)) };
+            gc.set_root();
+            gc
         }
     }
 }
 
+unsafe fn ptr<T: ?Sized + Trace>(ptr: NonZero<*const GcBox<T>>) -> NonZero<*const GcBox<T>> {
```

Comment:
Can we have a doc comment on this? It's use is not obvious from its name. Perhaps `remove_root_bit` would be a better name?